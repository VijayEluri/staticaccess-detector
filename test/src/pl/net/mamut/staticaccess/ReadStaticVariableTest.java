package pl.net.mamut.staticaccess;

/**
 * Reading a static variable test.
 * @author Adam Warski (adam at warski dot org)
 */
public class ReadStaticVariableTest {
	@StaticIndependent
	public void m1() {
	    String a = OtherClass.b;		// error
		method(OtherClass.b);			// error
	}

	@StaticIndependent
	public static void method(String param) { }

	public static void m2() {
		String a = OtherClass.b;
		method(OtherClass.b);	
	}

	private static class OtherClass {
		static String b = "10";
	}
}