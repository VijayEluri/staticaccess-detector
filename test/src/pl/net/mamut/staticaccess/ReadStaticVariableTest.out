SANA 10
SANA 11